### ISSUE TITLE
DON'T DELETE THIS CLOSING TAG -- uhh, why not?

### ISSUE THREAD

**Issue by dougcassidy** [eddiemachado/bones#795]
Title says it all, what gives?

**Comment by draivika** [comment]
Do you mean in the functions file? It is a warning so someone doesn't accidentally break the theme by removing the tag. I'm guessing it happened so often that Eddie decided it warranted the instructions.

**Comment by joshuaiz** [comment]
Yeah go ahead and remove it and see what happens. But it's there for a reason - your site will most likely break if you remove that tag. So that is why it's there.

**Comment by dougcassidy** [comment]
Yes, in the functions.php.  The closing ?> tag is not required, in fact, it's recommended that you dont put it at the end of a php file.  Are you guys aware of that?

**Comment by joshuaiz** [comment]
Oh I was referring to where it says that at the end of some functions like in bones.php.

**Comment by YemSalat** [comment]
@joshuaiz there are no such mentions in `bones.php`

Overall it is a good practice to omit the closing `?>` tag - because it prevents accidental whitespace output (like accidentally outputting whitespace before sending headers, etc.) 

I have no idea why it says "don't delete this closing tag" in functions.php either, would be good if someone could shine some light on it.
(I really hope this is not just another cargo cult thing that someone dragged along and nobody stopped them because they did not know better)

**Comment by joshuaiz** [comment]
@YemSalat you're right - there's no mention of it in bones.php. 

I was thinking of the `// don't remove this bracket!` in functions.php. My mistake.

And you are also right that you should remove whitespace output at the end of php files. I've ran into that problem a few times. 

So yes there is no reason not to remove that closing tag.
